packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -391,45 +404,141 @@ export default class API {
-        return cached.data as Metadata;
-      }
-      console.log(
-        '%c Checking for MetaData files',
-        'line-height: 30px;text-align: center;font-weight: bold',
-      );
-
-      const metadataRequestOptions = {
-        params: { ref: 'refs/meta/_netlify_cms' },
-        headers: { Accept: 'application/vnd.github.v3.raw' },
-        cache: 'no-store' as RequestCache,
-      };
+  async retrieveMetadataOld(key: string): Promise<Metadata> {
+    console.log(
+      '%c Checking for MetaData files',
+      'line-height: 30px;text-align: center;font-weight: bold',
+    );
 
-      const errorHandler = (err: Error) => {
-        if (err.message === 'Not Found') {
-          console.log(
-            '%c %s does not have metadata',
-            'line-height: 30px;text-align: center;font-weight: bold',
-            key,
-          );
-        }
-        throw err;
-      };
+    const metadataRequestOptions = {
+      params: { ref: 'refs/meta/_netlify_cms' },
+      headers: { Accept: 'application/vnd.github.v3.raw' },
+      cache: 'no-store' as RequestCache,
+    };
 
-      if (!this.useOpenAuthoring) {
-        return this.request(`${this.repoURL}/contents/${key}.json`, metadataRequestOptions)
-          .then((response: string) => JSON.parse(response))
-          .catch(errorHandler);
+    const errorHandler = (err: Error) => {
+      if (err.message === 'Not Found') {
+        console.log(
+          '%c %s does not have metadata',
+          'line-height: 30px;text-align: center;font-weight: bold',
+          key,
+        );
       }
+      throw err;
+    };
 
-      const [user, repo] = key.split('/');
-      return this.request(`/repos/${user}/${repo}/contents/${key}.json`, metadataRequestOptions)
+    if (!this.useOpenAuthoring) {
+      const result = await this.request(
+        `${this.repoURL}/contents/${key}.json`,
+        metadataRequestOptions,
+      )
         .then((response: string) => JSON.parse(response))
         .catch(errorHandler);
-    });
+
+      return result;
+    }
+
+    const [user, repo] = key.split('/');
+    const result = this.request(
+      `/repos/${user}/${repo}/contents/${key}.json`,
+      metadataRequestOptions,
+    )
+      .then((response: string) => JSON.parse(response))
+      .catch(errorHandler);
+    return result;
+  }
+
+  async getPullRequests(
+    head: string | undefined,
+    state: PullRequestState,
+    predicate: (pr: GitHubPull) => boolean,
+  ) {
+    const pullRequests: Octokit.PullsListResponse = await this.requestAllPages(
+      `${this.originRepoURL}/pulls`,
+      {
+        params: {
+          ...(head ? { head: await this.getHeadReference(head) } : {}),
+          base: this.branch,
+          state,
+          // eslint-disable-next-line @typescript-eslint/camelcase
+          per_page: 100,
+        },
+      },
+    );
+
+    return pullRequests.filter(
+      pr => pr.head.ref.startsWith(`${CMS_BRANCH_PREFIX}/`) && predicate(pr),
```

Comment:
@erquhart ``pr.head.ref.startsWith(`${CMS_BRANCH_PREFIX}/`)`` was missing from the previous PR.